tests/suites/cloud/tests/ssh-auth/index.js

```diff
@@ -122,78 +122,60 @@ module.exports = {
 						);
 					});
 				}).then(async () => {
-					return exec(`ssh-add ${sshPath}`)
-					.then(() => {
-						this.worker.addSSHKey(sshPath);
-					})
-					.then(() => {
-						return setConfig(test, this, this.balena.uuid, 'os.sshKeys', [customKey.pubKey.trim()]);
-					});
+						return setConfig(test, this, this.balena.uuid, 'os.sshKeys', [this.context.get().sshKey.pubKey.trim()]);
 				}).then(async () => {
 					let result;
-					let ip = await this.worker.getDutIp(this.link);
-					let config = {}
-					config = {
-						host: ip,
-						port: '22222',
-						username: 'root',
-						privateKeyPath: `${sshPath}`
-					};
 					await this.utils.waitUntil(
 						async () => {
-							try {
-								result = await this.utils.executeCommandOverSSH('echo -n pass',
-								config);
-							} catch (err) {
-								console.error(err.message);
-								throw new Error(err);
-							}
+							result = await this.worker.executeCommandInHostOS('echo -n pass',
+								this.link);
 							return result
 						}, false, 10, 5 * 1000);
 					return test.equals(
-						result.stdout,
+						result,
 						"pass",
 						"Local SSH authentication with custom keys is allowed in production mode"
 					);
 			  }).then(async () => {
 					await setConfig(test, this, this.balena.uuid, 'os.sshKeys');
 			  }).then(async () => {
 					let result;
+					let config = {};
 					let ip = await this.worker.getDutIp(this.link);
-					let config = {}
-					await this.cloud.balena.models.key.create(this.suite.options.id, customKey.pubKey);
-					config = {
-						host: ip,
-						port: '22222',
-						username: this.worker.username,
-						privateKeyPath: `${sshPath}`
-					};
 					await this.utils.waitUntil(
 						async () => {
-							try {
-								result = await this.utils.executeCommandOverSSH('echo -n pass',
-								config);
-							} catch (err) {
-								console.error(err.message);
-								throw new Error(err);
-							}
-							return result
+							if (!this.worker.directConnect) {
```

Comment:
does this mean this test will not be run on the qemu environment?

Reply:
hey @rcooke-warwick my intention was to run it both for testbot and qemu, as in:
```
147               if (!this.worker.directConnect) {                                 
                                           // This runs for a testbot device 
151                 } else {                                                        
                                           // This runs for a QEMU device              
163             }, false, 10, 5 * 1000);
```
Is that not how you understand it?